tests/gordo_components/model/anomaly/test_anomaly_detectors.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+
+from datetime import timedelta
+
+import pytest
+import numpy as np
+import pandas as pd
+import yaml
+
+from sklearn.decomposition import PCA
+from sklearn.preprocessing import MinMaxScaler, RobustScaler
+
+from gordo_components import serializer
+from gordo_components.model import utils as model_utils
+from gordo_components.model.anomaly import DiffBasedAnomalyDetector
+
+
+@pytest.mark.parametrize("base_estimator", (PCA(),))
+@pytest.mark.parametrize("scaler", (MinMaxScaler(), RobustScaler()))
+@pytest.mark.parametrize(
+    "index", (range(10), pd.date_range("2019-01-01", "2019-01-30", periods=10))
+)
+@pytest.mark.parametrize("lookback", (0, 5))
+def test_diff_detector(base_estimator, scaler, index, lookback):
+    """
+    Test the functionality of the DiffBasedAnomalyDetector
+    """
+
+    def simulate_lstm_predict(func):
```

Comment:
This is because using our actual lstm-implementation is too slow? Is it actually noticeable slower on this small dataset? If it is not then I think I would prefer using our actual keras lstm.

Reply:
It's more the `model.compile(...)` which takes the time. I can swap it, but it goes from running the 8 tests (parameterized) in ~1 second, to 2 minutes. :laughing:

Reply:
And that is not just because it is the first time you had to load in tensorflow in that session?

Reply:
Each test ends up taking 20 some odd seconds. And looking at the slowest tests reports, it's mostly LSTM type tests.